src/compiler/diagnosticMessages.json

```diff
@@ -1755,6 +1751,10 @@
         "category": "Error",
         "code": 2535
     },
+    "The 'arguments' object cannot be referenced in a generator function or method in ES3 and ES5. Consider using a standard function or method.": {

```

Comment:
The same argument could be made for arrow functions, which have pretty much the same error message. I'd rather not overcomplicate the error message here.